sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java

```diff
@@ -203,7 +219,7 @@ private HttpPipeline createHttpPipeline(HttpClient httpClient,
 
     private void applyRequiredPolicies(List<HttpPipelinePolicy> policies) {
         policies.add(getUserAgentPolicy());
-        policies.add(new RetryPolicy());
+        policies.add(this.retryPolicy == null ? new RetryPolicy() : this.retryPolicy);
```

Comment:
Does it make sense to have this.retryPolicy  set to new Retrypolicy() in the constructor of the class. And in the method `public ChatClientBuilder retryPolicy(RetryPolicy retryPolicy)`, it just reset the retryPolicy attribute
This way, we can avoid the null checks in the code.

Reply:
Yea, I think that would be possible. But I checked a few other SDKs and the common pattern is to do a null check like we have here. I would prefer to keep this consistent with other SDKs :)